lib/kits/core-ui/page-decorations.ts

```diff
@@ -42,14 +70,92 @@ const PageDecorations = () => {
         );
 
         filteredPageDecorations?.forEach((pageDecoration) => {
-            if (
+            if (pageDecoration && pageDecoration.hotspots?.length) {
+                pageDecoration.hotspots?.forEach((hotspot, index) => {
+                    const el = document.createElement('div');
+                    el.classList.add('sgn-pagedecoration-hotspot');
+                    el.classList.add(`sgn-pagedecoration-hotspot-${index}`);
+
+                    let x1 = hotspot.x1;
+                    let x2 = hotspot.x2;
+
+                    const versoPageSpread = versoPageSpreads?.find(
+                        (pageSpread) =>
+                            pageSpread.pageIds.includes(
+                                `page${pageDecoration.page_number}`
+                            )
+                    );
+                    const contentRect = versoPageSpread?.getContentRect();
+
+                    const pageSpreadEl = pageSpreads
```

Comment:
Couldn't we make it so that we simply add page decorations to the hotspots array before rendering? Then we kind of skip this code path.